app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/rejectedgoodsmultiple/EnterMovementReferenceNumberController.scala

```diff
@@ -46,78 +49,105 @@ class EnterMovementReferenceNumberController @Inject() (
         )
       )
     }
   }
 
-  def enterMrnSubmit(): Action[AnyContent] = actionReadWriteJourney { implicit request => journey =>
-    movementReferenceNumberForm
-      .bindFromRequest()
-      .fold(
-        formWithErrors =>
-          (
-            journey,
-            BadRequest(
-              enterMovementReferenceNumberPage(
-                formWithErrors,
-                ReimbursementRoutes(JourneyBindable.Multiple).subKey,
-                routes.EnterMovementReferenceNumberController.enterMrnSubmit()
+  def enterMrnSubmit(leadOrOrdinalValue: Int): Action[AnyContent] = actionReadWriteJourney {
+    implicit request => journey =>
+      movementReferenceNumberForm
+        .bindFromRequest()
+        .fold(
+          formWithErrors =>
+            (
+              journey,
+              BadRequest(
+                enterMovementReferenceNumberPage(
+                  formWithErrors,
+                  ReimbursementRoutes(JourneyBindable.Multiple).subKey,
+                  getOrdinalValue(leadOrOrdinalValue),
+                  routes.EnterMovementReferenceNumberController.enterMrnSubmit(leadOrOrdinalValue)
+                )
               )
-            )
-          ).asFuture,
-        mrnNumber =>
-          claimService
-            .getDisplayDeclaration(mrnNumber)
-            .fold(
-              errors => {
-                logger.error(s"Unable to record $mrnNumber", errors.toException)
-                (journey, Redirect(baseRoutes.IneligibleController.ineligible()))
-              },
-              {
-                case Some(acc14) =>
-                  journey
-                    .submitMovementReferenceNumberAndDeclaration(mrnNumber, acc14)
-                    .fold(
-                      error => {
-                        logger.error(s"Unable to update journey [$error]")
-                        (journey, Redirect(baseRoutes.IneligibleController.ineligible()))
-                      },
-                      updatedJourney => (updatedJourney, redirectLocation(updatedJourney))
-                    )
-                case None        =>
-                  logger.error(s"Display Declaration details not found")
+            ).asFuture,
+          mrn => {
+            val index = getOrdinalValue(leadOrOrdinalValue) - 1
+            claimService
+              .getDisplayDeclaration(mrn)
+              .fold(
+                errors => {
+                  logger.error(s"Unable to record $mrn", errors.toException)
                   (journey, Redirect(baseRoutes.IneligibleController.ineligible()))
-              }
-            )
-      )
+                },
+                {
+                  case Some(acc14) =>
+                    if (index =!= 0 && journey.newMrnEorisAndLeadMrnEorisMismatch(acc14)) {
+                      (journey, BadRequest(customError(mrn, leadOrOrdinalValue, "multiple.error.wrongMRN")))
+                    } else if (index =!= 0 && journey.movementReferenceNumberExists(mrn)) {
```

Comment:
this is already covered by `submitMovementReferenceNumberAndDeclaration`, see https://github.com/hmrc/cds-reimbursement-claim-frontend/blob/8252e064f7e471c4f3ee52e9335f4ac1e73405aa/app/uk/gov/hmrc/cdsreimbursementclaimfrontend/journeys/RejectedGoodsMultipleJourney.scala#L364-L364